fl4health/clients/basic_client.py

```diff
@@ -625,8 +626,12 @@ def predict(self, input: torch.Tensor) -> Tuple[Dict[str, torch.Tensor], Dict[st
             raise ValueError("Model forward did not return a tensor or dictionary or tensors")
 
     def compute_loss(
```

Comment:
I see some advantage to using the boolean here. If we do that, the user need only define one `compute_loss` function. 

However, I wonder what you think have having a `compute_training_loss` and `compute_eval_loss`. This would allow for more explicit typing and specific workflows, like only constructing backward or checkpoint when necessary. In the more complex overrides, code duplication could be reduce by defining a unified `compute_loss` function that both call, for example in `fenda_client.py`